src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultProjectWorkspaceStateGenerator.cs

```diff
@@ -24,18 +26,25 @@ internal class DefaultProjectWorkspaceStateGenerator : ProjectWorkspaceStateGene
 
         private readonly ProjectSnapshotManagerDispatcher _projectSnapshotManagerDispatcher;
         private readonly SemaphoreSlim _semaphore;
+        private readonly ITelemetryReporter _telemetryReporter;
         private ProjectSnapshotManagerBase _projectManager;
         private TagHelperResolver _tagHelperResolver;
         private bool _disposed;
 
         [ImportingConstructor]
-        public DefaultProjectWorkspaceStateGenerator(ProjectSnapshotManagerDispatcher projectSnapshotManagerDispatcher)
+        public DefaultProjectWorkspaceStateGenerator(ProjectSnapshotManagerDispatcher projectSnapshotManagerDispatcher, ITelemetryReporter telemetryReporter)
         {
             if (projectSnapshotManagerDispatcher is null)
             {
                 throw new ArgumentNullException(nameof(projectSnapshotManagerDispatcher));
             }
 
+            if (telemetryReporter is null)
+            {
+                throw new ArgumentNullException(nameof(telemetryReporter));
+            }
+
+            _telemetryReporter = telemetryReporter;
             _projectSnapshotManagerDispatcher = projectSnapshotManagerDispatcher;
```

Comment:
```suggestion
            _projectSnapshotManagerDispatcher = projectSnapshotManagerDispatcher ?? throw new ArgumentNullException(nameof(projectSnapshotManagerDispatcher));
            _telemetryReporter = telemetryReporter ?? throw new ArgumentNullException(nameof(telemetryReporter));
```

Reply:
I wasn't sure if we've decided to standardize on this. If so I'm happy to update everywhere I see doing this as I update files.

Reply:
If we decide to standardize then this should be part of editorconfig imo (if it's not there already)

Reply:
Yeah, we're inconsistent on this. It's used in some places and not used in others. I would expect that the more laborious forms existed before we allowed throw expressions in C#. I think it's fine to add this to the list of things to discuss in #6941.

At any rate, the reason I noticed it is because the null checks and field assignments are in a different order. It's a tiny nit, but would you mind tidying that up?